Ginkgo: Migrated headless service tests

eloycoto opened the pull request:
Migrated `04-toservices-test.sh` to Ginkgo test framework

Fix #1985

PS: I know that all these manifest files duplicated in two folders are not the best way. I have a small snippet for fixing this in a better way, but I need to test a bit deeper, so that will be fixed in the coming days. 


Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Ginkgo: Migrated headless service tests

Migrated `04-toservices-test.sh` to Ginkgo test framework

Fix #1985

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

ianvernon commented on test/helpers/kubectl.go:
```diff
@@ -105,6 +105,19 @@ func (kub *Kubectl) GetPods(namespace string, filter string) *CmdRes {
 }
 
+// GetPodsNodes returns a map with pod name as a key and node name as value. It
+// get only pods in the given namespace that match the provided filter.
```
It only gets

ianvernon commented on test/k8sT/Services.go:
```diff
@@ -146,4 +147,105 @@ var _ = Describe("K8sServicesTest", func() {
+				}
+				return false
+			}
+			err := helpers.WithTimeout(waitFinish, "cannot finish containers",
```
change to "cannot finish deleting containers"

ianvernon reviewed (approved):
Minor nitpicks, doesn't need another round of review from me. @nebril please confirm that this test adequately migrates the test from the bash scripts, as you are the author of the original test. Thanks!

nebril commented on test/k8sT/Services.go:
```diff
@@ -146,4 +147,105 @@ var _ = Describe("K8sServicesTest", func() {
+		})
+
+		It("To Services first policy", func() {
+			res := kubectl.Apply(podPath)
```
Please move pod creation to BeforeEach and ensure it is created to be consistent with original test behaviour.

nebril commented on test/k8sT/Services.go:
```diff
@@ -146,4 +147,105 @@ var _ = Describe("K8sServicesTest", func() {
+			res := kubectl.Apply(endpointPath)
+			res.ExpectSuccess()
+
+			res = kubectl.Apply(podPath)
```
Please move pod creation to BeforeEach and ensure it is created to be consistent with original test behaviour.

eloycoto replied:
Hi @nebril 

Quick one, I made like this that is how the original test is made. First endpoint -> Pods and after that the other way around.

https://github.com/cilium/cilium/blob/master/tests/k8s/tests/04-toservices-test.sh#L66
https://github.com/cilium/cilium/blob/master/tests/k8s/tests/04-toservices-test.sh#L97

nebril reviewed (changes requested):
Looks good overall!

Just need to move pod creation to BeforeEach (it's not part of the test, it's part of test scenario).

eloycoto committed:
Ginkgo: Fix PR comments

- Move pod creation to the beforeEach.
- Update the manifest to the new folder structure.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
